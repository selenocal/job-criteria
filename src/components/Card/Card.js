import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Card = ({ children, width }) => (
  <StyledCard width={width}>{children}</StyledCard>
);

const StyledCard = styled.div`
  background-color: #fff;
  padding: 20px;
  border: 1px solid #dddddd;
  border-radius: 4px;
  margin: 50px;
  width: ${props => (props.width ? props.width : "100%")};
`;

Card.propTypes = {
  width: PropTypes.string
};

Card.defaultProps = {
  width: "100%"
};

export default Card;
