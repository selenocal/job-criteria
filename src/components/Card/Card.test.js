import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("<Card />", () => {
  it("renders Card correctly", () => {
    const wrapper = shallow(
      <Card>
        <div>
          <span>Test</span>
        </div>
      </Card>
    );

    expect(wrapper.contains("Test")).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
