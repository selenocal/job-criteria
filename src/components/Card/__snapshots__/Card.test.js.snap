// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders Card correctly 1`] = `ShallowWrapper {}`;
